### ISSUE TITLE
Object is not a hash

### ISSUE THREAD

**Issue by gaearon** [gaearon/flux-react-router-example#16]
[An object is not a hash.](http://www.devthought.com/2012/01/18/an-object-is-not-a-hash/)

Currently example fails with usernames such as `constructor`.

We solve this partially with https://github.com/gaearon/flux-react-router-example/commit/487745ffe9eddc06cf37001ecd6c9735c0d640b2 but the same needs to be done for `RepoStore` and `UserStore`. The easiest way to do this is to create a `Bag` class and use it instead of `mergeIntoBag`, `isInBag` helpers. Inside `Bag`, we will use prefixes for IDs.

**Comment by JakeChampion** [comment]
Could ES6 Maps be used for this instead?

**Comment by gaearon** [comment]
Yes but they're not really there yet. I don't know much about the polyfill though.

**Comment by JakeChampion** [comment]
[Kangax's ES6 compat table](https://kangax.github.io/compat-table/es6/#babel) shows that Babel passes all the Map tests. The polyfill is a part of [core-js](https://github.com/zloirock/core-js#map).
